src/main/java/org/myrobotlab/service/InMoov.java

```diff
@@ -154,9 +161,39 @@ public InMoov(String n) {
   // ---------------------------------------------------------------
 
   public void attach(Attachable attachable) {
+    //opencv
     if (attachable instanceof OpenCV) {
       opencv = (OpenCV) attachable;
+      subscribe(opencv.getName(), "publishClassification");
+      //SpeechRecognizer
+    } else if (attachable instanceof SpeechRecognizer) {
+      if (attachable.getClass().getSimpleName().equals("WebkitSpeechRecognition")) {
+        WebGui webgui = (WebGui) Runtime.create("webgui", "WebGui");
+        webgui.autoStartBrowser(false);
+        webgui.startService();
+        webgui.startBrowser("http://localhost:8888/#/service/" + attachable.getName());
+      }
+      ear = (SpeechRecognizer) attachable;
+    } else if (attachable instanceof ProgramAB) {
+      if (htmlFilter == null) {
+        htmlFilter = (HtmlFilter) Runtime.start(this.getIntanceName() + ".htmlFilter", "HtmlFilter");
+      }
+      chatBot = (ProgramAB) attachable;
+    }
+    //sub attach
+    if (chatBot != null && ear != null) {
+      chatBot.attach((Attachable) ear);
+    }
+    if (chatBot != null && mouth != null) {
+      //TODO attach...
+      chatBot.addTextListener(htmlFilter);
```

Comment:
this should be done in python as it was before.  Changes like this are for InMoov2 in ogre not Nixie!

Reply:
we can do it as it was before, just added a shorcut ... but reverted ...